docs/types.rst

```diff
@@ -69,6 +69,28 @@ To mypy, this code is equivalent to the one above:
       a_number = attr.ib(default=42)  # type: int
       list_of_numbers = attr.ib(factory=list, type=typing.List[int])
 
+
+pyright
+-------
+
+``attrs`` provides support for pyright_ though the ``dataclass_transform`` specification.
+This provides static type inference for a subset of ``attrs`` equivalent to standard-library ``dataclasses``,
+and requires explicit type annotations using the :ref:`next-gen` or ``@attr.s(auto_attribs=True)`` APIs.
+Given the following definition, ``pyright`` will generate static type signatures for ``SomeClass``,
+``SomeClass`` properties, ``SomeClass.__init__``, and ``SomeClass`` comparision methods.
```

Comment:
Can we add notes for frozen, converters (both not supported at this time, IIUC) and wrapping `@define` (users wrapping need to also apply the `dataclass_transform` thing)?

Reply:
Absolutely, I see needs for:

- [ ] Notes on for frozen, converters here in the "provisional support" note.
- [ ] Notes on wrapping custom decorators in `docs/extending.rst`, with a minor refactor to separate out `pyright` and `mypy` type support.

I will comment back here once I begin this draft. As we're on a tight-ish schedule please feel free to jump in before then and draft (probably via a PR against this upstream branch, so we can keep this PR's comment thread).

Reply:
I've added draft docs.